File: Documentation Code Snippets/ScanbotSDK Examples/TIFF Creation/Objc/TIFFCreation.m
//
//  TIFFCreation.m
//  ScanbotSDK Examples
//
//  Created by Danil Voitenko on 26.10.21.
//

#import "TIFFCreation.h"
@import ScanbotSDK;

@implementation TIFFCreation

- (void)createTIFF {
    
    // For this example we're using an empty array, but there should be scanned images in it.
    NSArray<UIImage *> *scannedImages = @[];

    // Specify the file URL where the TIFF will be saved to. Nil makes no sense here.
    NSURL *outputURL = [[NSURL alloc] initWithString:@"outputTIFF"];

    // In case you want to encrypt your TIFF file, create encrypter using a password and an encryption mode.
    SBSDKAESEncrypter *encrypter = [[SBSDKAESEncrypter alloc] initWithPassword:@"password_example#42"
                                                                          mode:SBSDKAESEncrypterModeAES256];

    // The `SBSDKTIFFImageWriter` has parameters where you can define various options,
    // e.g. compression algorithm or whether the document should be binarized.
    // For this example we're going to use the default parameters.
    SBSDKTIFFImageWriterParameters *parameters = [SBSDKTIFFImageWriterParameters defaultParameters];
    
    // Create the tiff image writer using created parameters and the encrypter.
    SBSDKTIFFImageWriter *tiffImageWriter = [[SBSDKTIFFImageWriter alloc] initWithParameters:parameters
                                                                                   encrypter:encrypter];

    // Write a TIFF file with scanned images into the defined URL.
    // The completion handler passes a file URL where the file was to be saved, or nil if the operation did not succeed.
    [tiffImageWriter writeTIFFWith:scannedImages toFile:outputURL completion:^(NSURL * _Nullable url) {
        
        // Handle the URL.
    }];
}

@end

File: Documentation Code Snippets/ScanbotSDK Examples/TIFF Creation/Swift/TIFFCreation.swift
//
//  TIFFCreation.swift
//  ScanbotSDK Examples
//
//  Created by Danil Voitenko on 26.10.21.
//

import Foundation
import ScanbotSDK

func createTIFF(from scannedDocument: SBSDKScannedDocument) {
    // The `SBSDKTIFFImageWriter` has parameters where you can define various options,
    // e.g. compression algorithm or whether the document should be binarized.
    // For this example we're going to use the default parameters.
    let parameters = SBSDKTIFFImageWriterParameters.defaultParameters
    
    // Create the tiff image writer using created parameters.
    let writer = SBSDKTIFFImageWriter(parameters: parameters)
    
    // Synchronously converts the scanned document to a multipage-TIFF file and writes it to the specified URL.
    //If output URL is `nil`the default TIFF location of the scanned document will be used.
    writer.writeTIFF(scannedDocument: scannedDocument)
}

func createTIFF(from images: [UIImage]) {
    
    // Specify the file URL where the TIFF will be saved to. Nil makes no sense here.
    guard let outputTIFFURL = URL(string: "outputTIFF") else { return }

    // In case you want to encrypt your TIFF file, create encrypter using a password and an encryption mode.
    guard let encrypter = SBSDKAESEncrypter(password: "password_example#42",
                                            mode: .AES256) else { return }
    
    // The `SBSDKTIFFImageWriter` has parameters where you can define various options,
    // e.g. compression algorithm or whether the document should be binarized.
    // For this example we're going to use the default parameters.
    let parameters = SBSDKTIFFImageWriterParameters.defaultParameters
    
    // Create the tiff image writer using created parameters and the encrypter.
    let tiffImageWriter = SBSDKTIFFImageWriter(parameters: parameters, encrypter: encrypter)

    // Asynchronously writes a TIFF file with scanned images into the defined URL.
    // The completion handler passes a file URL where the file was to be saved, or nil if the operation did not succeed.
    tiffImageWriter.writeTIFF(with: images, toFile: outputTIFFURL, completion: { url in
        
        // Handle the URL.
    })
}